asmcomp/compilenv.mli

```diff
@@ -14,7 +14,7 @@
 
 open Cmx_format
 
-val reset: ?packname:string -> string -> unit
+val reset: ?packname:string -> sourcefile:Timings.build_kind -> string -> unit
```

Comment:
Looking at the type definitions for `part` and `build_kind` it is clear that:

- `part` indicates what the compiler is doing with the code it is currently processing; it corresponds to the usual notion of "compiler pass".
- `build_kind` indicate where the code that is being processed comes from; I would call this "source provenance" (builtin startup code being a provenance distinct from "the code from [this file]").

You should never use the argument name `sourcefile` for something that is not a "source file". If it was my code, I would:

- rename `part` into `compiler_pass`
- rename `build_kind` into `source_provenance`
- label `source_provenance` arguments with the name `source_provenance`